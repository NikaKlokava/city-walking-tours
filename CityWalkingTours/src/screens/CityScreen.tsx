import React, {useState} from 'react';
import {Image, ScrollView, StyleSheet, TextInput, View} from 'react-native';
import {CITIES, SECTIONS} from '../utils/data';
import {Text} from '../components/base/Text';
import {colors} from '../utils/colors';
import {flexRow} from '../utils/flex';
import {Line} from '../components/Line';
import {SearchBar} from '../components/SearchBar';
import {BackBtn} from '../components/BackBtn';
import {Categories} from '../components/Categories';
import {SelectedCategory} from '../components/SelectedCategory';
import {Icon} from '../components/base/Icon';

const icon = require('../assets/search.png');

export const CityScreen = () => {
  const [selectedCategory, setSelectedCategory] = useState<CategotyType>();

  const handleCategorySelect = (title: string) => {
    const category = SECTIONS.find(section => section.title === title);
    setSelectedCategory(category);
  };
  return (
    <View style={styles.container}>
      <View style={[styles.headerContainer, flexRow]}>
        <BackBtn />
        <View style={[styles.cityContainer, flexRow]}>
          <Text
            type="tertiary"
            center
            color={colors.primary1}
            style={styles.opacity}>
            city:
          </Text>
          <Text type="primary" center color={colors.primary1}>
            {CITIES[0].city}
          </Text>
        </View>
      </View>
      <Line />
      <ScrollView>
        <View style={styles.scrollContainer}>
          {!selectedCategory && (
            <View style={[styles.inputContainer, flexRow]}>
              <TextInput
                style={styles.searchInput}
                placeholder="What do you want to find?"
                placeholderTextColor={`rgba(250,250,250,0.7)`}
              />
              <Icon source={icon} size="medium" />
            </View>
          )}
          <SearchBar
            onSelect={handleCategorySelect}
            title={selectedCategory?.title}
          />
          {selectedCategory ? (
            <SelectedCategory category={selectedCategory} />
          ) : (
            <Categories categories={SECTIONS} onSelect={handleCategorySelect} />
          )}
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 80,
    paddingHorizontal: 10,
  },
  headerContainer: {
    justifyContent: 'space-between',
    marginBottom: 20,
  },
  scrollContainer: {rowGap: 20, marginTop: 10},
  cityContainer: {columnGap: 10},
  inputContainer: {
    padding: 15,
    backgroundColor: '#263238', // !!!
    borderRadius: 10,
    justifyContent: 'space-between',
  },
  searchInput: {
    color: colors.primary1,
    fontSize: 15,
  },
  opacity: {
    opacity: 0.7,
  },
});
