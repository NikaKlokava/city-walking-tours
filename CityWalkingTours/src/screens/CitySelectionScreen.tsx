import React, {useState} from 'react';
import {ImageBackground, StyleSheet, View} from 'react-native';
import {StyledBtn} from '../components/StyledBtn';
import {Text} from '../components/base/Text';
import {colors} from '../utils/colors';
import {SelectCityModal} from '../components/SelectCityModal';
import {AppWrapper} from '../components/AppWrapper';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';

const image = require('../assets/city.png');

type Props = {
  navigation: NativeStackNavigationProp<any, any>;
};

export const CitySelectionScreen = ({navigation}: Props) => {
  const [modalVisible, setModalVisible] = useState(false);
  const [currentCity, setCurrentCity] = useState<string>('');

  const handleModalClose = () => {
    setModalVisible(false);
  };

  const handleSelectCity = (city: string) => {
    setCurrentCity(city);
    setModalVisible(false);
  };

  return (
    <AppWrapper>
      <ImageBackground source={image} style={styles.container}>
        {!modalVisible && (
          <>
            <View style={styles.selectBlock}>
              <Text type="primary" color={colors.primary3} center>
                Select your city
              </Text>
              <StyledBtn title="SELECT" onClick={() => setModalVisible(true)} />
              {currentCity && (
                <Text type="primary" color={colors.primary3} center>
                  {currentCity}
                </Text>
              )}
            </View>
            {currentCity && (
              <StyledBtn
                title="NEXT"
                onClick={() => navigation.navigate('City', {city: currentCity})}
              />
            )}
          </>
        )}
      </ImageBackground>
      <SelectCityModal
        visible={modalVisible}
        onClose={handleModalClose}
        onSelect={handleSelectCity}
      />
    </AppWrapper>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 80,
    justifyContent: 'space-between',
  },
  selectBlock: {
    rowGap: 60,
  },
});
