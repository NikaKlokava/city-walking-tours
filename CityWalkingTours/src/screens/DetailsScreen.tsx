import React, {useRef} from 'react';
import {
  Animated,
  ImageBackground,
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';
import {Text} from '../components/base/Text';
import {commonStyles, colors, PLACE, DEVICE_HEIGHT} from '../utils';
import {BackBtn} from '../components/BackBtn';
import {Rating} from '../components/Rating';
import {Line} from '../components/Line';
import {Icon} from '../components/base/Icon';
import {Details} from '../components/Details';
import {Gallery} from '../components/Gallery';
import {AppWrapper} from '../components/AppWrapper';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import SVG_LOCATION from '../assets/icons/location.svg';

const image = require('../assets/images/sodas.png');

type Props = {
  navigation: NativeStackNavigationProp<any, any>;
};

const MAX_HEADER_HEIGHT = DEVICE_HEIGHT * 0.4;
const MIN_HEADER_HEIGHT = DEVICE_HEIGHT * 0.3;
const SCROLL_DiISTANCE = MAX_HEADER_HEIGHT - MIN_HEADER_HEIGHT;

export const DetailsScreen = ({navigation}: Props) => {
  const scrollOffsetY = useRef(new Animated.Value(0)).current;

  const animatedHeaderHeight = scrollOffsetY.interpolate({
    inputRange: [0, SCROLL_DiISTANCE],
    outputRange: [MAX_HEADER_HEIGHT, MIN_HEADER_HEIGHT],
    extrapolate: 'clamp',
  });

  const onHandleScroll = () => {
    Animated.event([{nativeEvent: {contentOffset: {y: scrollOffsetY}}}], {
      useNativeDriver: false,
    });
  };

  return (
    <AppWrapper noPaddingTop>
      <View style={StyleSheet.flatten([commonStyles.container])}>
        <Animated.View style={{height: animatedHeaderHeight}}>
          <ImageBackground source={image} style={styles.headerContainer}>
            <View style={styles.backBtnContainer}>
              <BackBtn onClick={() => navigation.goBack()} />
            </View>
            <View style={styles.headerDescription}>
              <Text type="primary" color={colors.primary1}>
                {PLACE.title}
              </Text>
              <View
                style={StyleSheet.flatten([
                  styles.locationContainer,
                  commonStyles.flexRow,
                ])}>
                <Icon icon={SVG_LOCATION} size="small" />
                <Text type="quaternary" color={colors.primary1}>
                  {PLACE.details.location}
                </Text>
              </View>
              <Rating rating={PLACE.rating} white />
            </View>
          </ImageBackground>
        </Animated.View>

        <ScrollView scrollEventThrottle={16} onScroll={onHandleScroll}>
          <View style={styles.generalInfo}>
            <Line white />
            <Text type="tertiary" color={colors.semi_primary1}>
              {PLACE.description}
            </Text>
            <Line />
          </View>
          <View style={styles.detailsContainer}>
            <Details title={PLACE.details.location} type="location" />
            <Details title={PLACE.details.workingHours} type="hours" />
            <Details title={PLACE.details.site} type="site" />
          </View>
          <Gallery />
        </ScrollView>
      </View>
    </AppWrapper>
  );
};

const styles = StyleSheet.create({
  headerContainer: {
    height: '100%',
    justifyContent: 'space-between',
  },
  backBtnContainer: {
    marginTop: 70,
    marginLeft: 20,
    opacity: 0.7,
  },
  headerDescription: {
    backgroundColor: colors.semi_primary2,
    paddingHorizontal: 10,
    paddingVertical: 5,
    rowGap: 5,
  },
  locationContainer: {columnGap: 5},
  generalInfo: {
    paddingHorizontal: 10,
    marginTop: 20,
    rowGap: 20,
  },
  detailsContainer: {margin: 30, rowGap: 20},
});
