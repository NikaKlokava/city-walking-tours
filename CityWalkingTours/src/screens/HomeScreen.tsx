import React from 'react';
import {Image, StyleSheet, Text, View, Dimensions} from 'react-native';
import {StyledBtn} from '../components/StyledBtn';

const image = require('../assets/back_travel_img.png');
const windowWidth = Dimensions.get('window').width;

export const HomeScreen = () => {
  return (
    <View style={styles.main}>
      <View style={styles.container}>
        <Image source={image} style={styles.backImg} />
        <View>
          <Text style={styles.title}>Escape the ordinary life</Text>
          <Text style={styles.description}>
            Discover great experiences around you and make you live interesting!
          </Text>
        </View>
      </View>
      <StyledBtn title="Get Started" />
    </View>
  );
};
const styles = StyleSheet.create({
  main: {
    flex: 1,
  },
  container: {
    flex: 3,
    rowGap: 20,
  },
  backImg: {
    width: 0.9 * windowWidth,
    height: 300,
    alignSelf: 'center',
  },
  title: {
    fontSize: 26,
    color: 'white',
    textAlign: 'center',
    fontFamily: 'Georgia',
    marginBottom: 10,
  },
  description: {
    color: '#D6D2D2',
    textAlign: 'center',
    marginHorizontal: 20,
    fontSize: 16,
    letterSpacing: 1,
    fontFamily: 'Georgia',
  },
});
