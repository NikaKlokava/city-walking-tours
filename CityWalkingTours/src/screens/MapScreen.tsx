import React, {useEffect, useState} from 'react';
import {AppWrapper} from '../components/AppWrapper';
import {StyleSheet, View} from 'react-native';
import {GEOLOCATION_DATA, colors, commonStyles} from '../utils';
import MapView, {Callout, Marker} from 'react-native-maps';
import Geolocation from '@react-native-community/geolocation';
import {
  useNavigation,
  ParamListBase,
  NavigationProp,
} from '@react-navigation/native';
import {Text} from '../components/base/Text';
import {routes} from '../navigation';

type LocationType = {
  latitude: number;
  longitude: number;
  altitude: number | null;
  accuracy: number;
  altitudeAccuracy: number | null;
  heading: number | null;
  speed: number | null;
};

export const MapScreen = () => {
  const navigation: NavigationProp<ParamListBase> = useNavigation();

  const [currentLocation, setCurrentLocation] = useState<LocationType>();

  useEffect(() => {
    Geolocation.getCurrentPosition(
      position => setCurrentLocation(position.coords),
      error => console.log(error),
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000},
    );
  }, []);

  return (
    <AppWrapper noPaddingTop>
      <View style={StyleSheet.flatten([commonStyles.container])}>
        <MapView
          style={StyleSheet.flatten([commonStyles.container])}
          initialRegion={{
            latitude: 54.687157,
            longitude: 25.279652,
            latitudeDelta: 0.0955,
            longitudeDelta: 0.0421,
          }}>
          {currentLocation && (
            <Marker
              coordinate={{
                latitude: currentLocation.latitude,
                longitude: currentLocation.longitude,
              }}
              image={{uri: 'geolocation'}}
              title="Your Location"
              description="You here now"
            />
          )}
          {GEOLOCATION_DATA.map((item, index) => (
            <Marker
              key={index}
              coordinate={{latitude: item.latitude, longitude: item.longitude}}
              image={{uri: 'location'}}
              title="Location">
              <Callout
                style={styles.description}
                onPress={() => navigation.navigate(routes.DETAILS)}>
                <Text type="quaternary" color={colors.semi_primary2}>
                  Read more...
                </Text>
              </Callout>
            </Marker>
          ))}
        </MapView>
      </View>
    </AppWrapper>
  );
};

const styles = StyleSheet.create({
  description: {
    // padding: 10,
    // backgroundColor: 'pink',
  },
});
