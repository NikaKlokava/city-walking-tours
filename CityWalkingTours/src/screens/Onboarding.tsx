import React, {useEffect, useState} from 'react';
import {
  FlatList,
  Image,
  NativeScrollEvent,
  NativeSyntheticEvent,
  StyleSheet,
  View,
} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {StyledBtn} from '../components/StyledBtn';
import {
  DEVICE_HEIGHT,
  DEVICE_WIDTH,
  DATA,
  colors,
  getIndex,
  commonStyles,
} from '../utils';
import {Text} from '../components/base/Text';
import {Line} from '../components/Line';
import {ProgressBar} from '../components/ProgressBar';
import {AppWrapper} from '../components/AppWrapper';

type Props = {
  onSubmit: () => void;
};

export const Onboarding = ({onSubmit}: Props) => {
  const [currentIndex, setCurrentIndex] = useState<number>(0);

  const handleScreenScroll = (e: NativeSyntheticEvent<NativeScrollEvent>) => {
    const index = getIndex(e);
    setCurrentIndex(index);
  };

  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <AppWrapper>
      <View
        style={StyleSheet.flatten([
          styles.mainContainer,
          commonStyles.container,
        ])}>
        <View style={StyleSheet.flatten([commonStyles.container])}>
          <FlatList
            data={DATA}
            horizontal
            pagingEnabled
            onScroll={e => handleScreenScroll(e)}
            keyExtractor={(_, index) => index.toString()}
            showsHorizontalScrollIndicator={false}
            renderItem={({item}) => {
              return (
                <View style={styles.listContainer}>
                  <Image source={item.image} style={styles.backImg} />
                  <View style={styles.descriptionContainer}>
                    <Text type="primary" color={colors.primary1} center>
                      {item.title}
                    </Text>
                    <Line />
                    <Text type="tertiary" color={colors.semi_grey} center>
                      {item.description}
                    </Text>
                  </View>
                </View>
              );
            }}
          />
        </View>
        <View
          style={StyleSheet.flatten([
            styles.dottsContainer,
            commonStyles.flexRow,
          ])}>
          {currentIndex === DATA.length - 1 ? (
            <StyledBtn
              title="Get Started"
              onClick={() => {
                onSubmit();
                // save that onboarding is passed to the async storage
                // write TRUE to some variable in the async storage
              }}
            />
          ) : (
            <ProgressBar index={currentIndex} dataLength={DATA.length} />
          )}
        </View>
      </View>
    </AppWrapper>
  );
};
const styles = StyleSheet.create({
  mainContainer: {
    marginTop: 110,
  },
  listContainer: {
    width: DEVICE_WIDTH,
    rowGap: 70,
  },
  backImg: {
    height: 0.3 * DEVICE_HEIGHT,
    width: DEVICE_WIDTH,
    objectFit: 'contain',
  },
  descriptionContainer: {paddingHorizontal: 15, rowGap: 12},
  dottsContainer: {
    columnGap: 20,
    justifyContent: 'center',
  },
});
