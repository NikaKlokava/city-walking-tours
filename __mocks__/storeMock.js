export const settingsStoreMock = {
  city: null,
  cityUid: null,
  isOnboardingPassed: '',
  isLoading: true,
  uploadSettingsData: async () => Promise,
  updateCity: () => Promise,
  updateOnboarding: async () => Promise,
};

export const cityStoreMock = {
  data: [],
  isLoading: false,
  uploadCitiesData: () => PromiseConstructor,
  setData: () => null,
  setIsLoading: () => null,
};

export const sectionsStoreMock = {
  categories: [],
  data: [],
  isLoading: false,
  uploadData: async () => Promise,
  updateLikeStatus: () => Promise,
};
