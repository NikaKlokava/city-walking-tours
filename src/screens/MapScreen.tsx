import React, {useEffect, useState} from 'react';
import {AppWrapper} from '../components/AppWrapper';
import {Image, StyleSheet, View} from 'react-native';
import {INITIAL_REGION, colors, commonStyles} from '../utils';
import MapView, {Callout, Marker} from 'react-native-maps';
import Geolocation from '@react-native-community/geolocation';
import {
  useNavigation,
  ParamListBase,
  NavigationProp,
} from '@react-navigation/native';
import {Text} from '../components/base/Text';
import {routes} from '../navigation';
import {observer} from 'mobx-react';
import {sectionsStore} from '../context/sections-store';
import {Rating} from '../components/Rating';
import {Line} from '../components/Line';
import {useThemeContext} from '../context/theme-context';

type LocationType = {
  latitude: number;
  longitude: number;
  altitude: number | null;
  accuracy: number;
  altitudeAccuracy: number | null;
  heading: number | null;
  speed: number | null;
};

export const MapScreenContent = observer(({store}: {store: SectionsStore}) => {
  const navigation: NavigationProp<ParamListBase> = useNavigation();
  const {theme} = useThemeContext();

  const [currentLocation, setCurrentLocation] = useState<LocationType>();

  useEffect(() => {
    Geolocation.getCurrentPosition(
      position => setCurrentLocation(position.coords),
      error => console.log(error),
      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000},
    );
  }, []);

  return (
    <AppWrapper noPaddingTop>
      <View style={StyleSheet.flatten([commonStyles.container])}>
        <MapView
          style={StyleSheet.flatten([commonStyles.container])}
          initialRegion={INITIAL_REGION}>
          {currentLocation && (
            <Marker
              coordinate={{
                latitude: currentLocation.latitude,
                longitude: currentLocation.longitude,
              }}
              image={{uri: 'geolocation'}}
              title="Your Location"
              description="You here now"
            />
          )}
          {store.data.map(section =>
            section.data.map((place, index) => {
              return (
                <Marker
                  key={index}
                  coordinate={{
                    latitude: place.details.coordinates.latitude,
                    longitude: place.details.coordinates.longitude,
                  }}
                  image={{uri: 'location'}}
                  title="Location">
                  <Callout
                    style={styles.description}
                    onPress={() => navigation.navigate(routes.DETAILS, place)}>
                    <Image src={place.image} style={styles.imageStyle} />
                    <Text type="quaternary" color={theme.colors.primary2}>
                      {place.title}
                    </Text>
                    <Rating rating={place.rating} />
                    <Line black />
                    <Text type="quaternary" color={theme.colors.semi_primary2}>
                      Read more...
                    </Text>
                  </Callout>
                </Marker>
              );
            }),
          )}
        </MapView>
      </View>
    </AppWrapper>
  );
});

export const MapScreen = () => <MapScreenContent store={sectionsStore} />;

const styles = StyleSheet.create({
  description: {
    padding: 5,
    width: 230,
    height: 250,
  },
  imageStyle: {
    width: '100%',
    height: '60%',
  },
});
