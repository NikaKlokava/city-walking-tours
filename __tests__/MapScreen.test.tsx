import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {it, describe, expect} from '@jest/globals';
import {MapScreenContent} from '../src/screens/MapScreen';
import {sectionsStoreMock} from '../__mocks__/storeMock';

describe('Render MapScreen', () => {
  it('MapScreen renders correctly', () => {
    const MapScreenComponent = renderer.create(
      <MapScreenContent store={sectionsStoreMock} />,
    );
    expect(MapScreenComponent).toMatchSnapshot();
  });
});
