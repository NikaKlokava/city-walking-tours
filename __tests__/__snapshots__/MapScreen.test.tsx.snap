// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MapScreen MapScreen renders correctly 1`] = `
<BVLinearGradient
  colors={
    [
      4278394667,
      4278394667,
      4278190080,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "flex": 1,
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <ActivityIndicator
      color="#FF677E"
      size="large"
      style={
        {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      color="#FF677E"
      style={
        {
          "color": "#FF677E",
          "fontFamily": "Montserrat Alternates",
          "fontSize": 25,
        }
      }
      type="primary"
    >
      Loading...
    </Text>
  </View>
</BVLinearGradient>
`;
